source/text/typesetter.ts

```diff
@@ -19,199 +26,273 @@ import { Label } from './label';
+            if (!isDelimiter /* includes LineFeed */) {
+                continue;
+            }
+
+            if (currentWordIndex < i) {
+                /* Add previous word fragment (indicated by word index below current index). */
+                fragments.push([currentWordIndex, i, Typesetter.FragmentType.Word]);
+            }
+            const type = label.lineFeedAt(i) ? Typesetter.FragmentType.LineFeed : Typesetter.FragmentType.Delimiter;
+            fragments.push([i, i + 1, type]);
+            currentWordIndex = i + 1;
+        }
+        /* Account for last fragment that does not end with delimiter. */
+        if (!isDelimiter!) {
+            fragments.push([currentWordIndex, label.length, Typesetter.FragmentType.Word]);
         }
+
+        return fragments;
     }
 
     /**
-     * Adjusts the vertices for line anchor (done due after typesetting) w.r.t. the targeted anchoring.
-     * @param label - Label to adjust the y-positions for.
-     * @param glyphs - Glyph vertices for rendering to align the origins' y-components of (expected untransformed).
-     * @param begin - Vertex index to start alignment at.
-     * @param end - Vertex index to stop alignment at.
-     *
-     * @todo Apply once at the beginning! Initial offset!
+     * Compute fragment widths without kernings w.r.t. preceding and subsequent fragments.
+     * @param fragments -
+     * @param advances -
+     * @param kernings -
+     * @returns -
      */
-    protected static transformLineAnchor(label: Label,
-        glyphs: GlyphVertices | undefined, begin: number, end: number): void {
-        if (glyphs === undefined) {
-            return;
+    private static fragmentWidths(fragments: Array<Fragment>,
+        advances: Float32Array, kernings: Float32Array): Float32Array {
+
+        const widths = new Float32Array(fragments.length);
+        for (let i = 0; i < fragments.length; ++i) {
+            const fragment = fragments[i];
+            widths[i] = advances.subarray(fragment[0], fragment[1]).reduce((width, advance, index) =>
+                width + advance + (index < fragment[1] ? kernings[index] : 0.0), 0.0);
         }
+        return widths;
+    }
 
-        let offset = 0.0;
-        switch (label.lineAnchor) {
-            case Label.LineAnchor.Ascent:
-                offset = label.fontFace!.ascent;
-                break;
-            case Label.LineAnchor.Center:
-                offset = label.fontFace!.size * 0.5 + label.fontFace!.descent;
-                break;
-            case Label.LineAnchor.Descent:
-                offset = label.fontFace!.descent;
-                break;
-            case Label.LineAnchor.Top:
-                offset = label.fontFace!.base;
-                break;
-            case Label.LineAnchor.Bottom:
-                offset = label.fontFace!.base - label.fontFace!.lineHeight;
+    /**
+     * Computes the left and right side thresholds for elide computation.
```

Comment:
At some point, there should be an explanation of what 'elide' means. What do we elide, and what for?